src/RpcClient.ts

```diff
@@ -227,6 +234,15 @@ export class PostMessageRpcClient extends RpcClient {
         });
     }
 }
+/* tslint:disable-next-line:no-namespace */
+namespace PostMessageRpcClient {
+    export const enum ConnectionState {
+        DISCONNECTED,
+        CONNECTING,
+        CONNECTED,
+    }
```

Comment:
Would it be possible to put the `enum` inside the class as e.g. a `private static const enum ConnectionState` or something? Instead of having to declare it outside the class and then having to create a namespace and then having to export the class after the fact?

Reply:
I think that's generally not possible in Typescript.
That's why I started using this declaration merging pattern.
@Bettelstab ?

Reply:
No other way atm I am aware of.